File: python/CRUD-examples/select/bind_variables.py
# Code Sample from the tutorial at https://learncodeshare.net/2015/06/02/select-crud-using-cx_oracle/
#  section titled "Select specific rows"
# Using the base template, the example code executes a simple query using named bind variables,
#  uses fetchall to retrieve the data and displays the results.

import cx_Oracle
import os
connectString = os.getenv('DB_CONNECT') # The environment variable for the connect string: DB_CONNECT=user/password@database
con = cx_Oracle.connect(connectString)

# Query for Kim
cur = con.cursor()
person_name = 'Kim'
statement = 'select id, name, age, notes from lcs_people where name = :name'
cur.execute(statement, {'name':person_name})
res = cur.fetchall()
print (res)

File: ruby/CRUD-examples/select/bind_variables.rb
# The example code below executes a simple query using named bind variables,
#  uses fetch to retrieve the data and displays the results.
# Code Sample from the tutorial at https://learncodeshare.net/2016/09/09/select-crud-using-ruby-oci8/
#  section titled "Select specific rows"

require 'oci8'
connectString = ENV['DB_CONNECT'] # The environment variable for the connect string: DB_CONNECT=user/password@database
con = OCI8.new(connectString)

# Example code showing a simple query using named bind variables.
person_name = 'Kim'
statement = 'select id, name, age, notes from lcs_people where name=:name'
cursor = con.parse(statement)
cursor.bind_param('name', person_name)
cursor.exec
cursor.fetch do |row|
  printf "Id: %d, Name: %s, Age: %d, Notes: %s\n", row[0], row[1], row[2], row[3]
end
# End Example
